### PR TITLE
Refreshed introduction of Oríon's features

### PR DESCRIPTION
**Pull request by Thomsch**
This pull request is meant to be a discussion about how we can improve the current list of features to serve as a presentation of Oríon's moto and strong points (or core concepts). 

The current description is too factual (i.e., the information is too specific for an introduction aimed at new users) and somewhat confusing (e.g., some wording, redundant information with respect to the README).

My proposition revisits the moto to be more specific and impactful for newcomers but also provides a concrete design stance. Additionally, I refreshed the description items to be more engaging and to also give an overview of the strong points of our different features. Finally, I renamed the section because it was never a list of features, but more of a showcase.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -52,31 +52,17 @@ Check out our `pytorch example`_ for a quick demonstration!
...
-
-.. _MongoDB: https://www.mongodb.com/
-.. _Skopt: https://scikit-optimize.github.io/
-.. _plugin documentation: https://orion.readthedocs.io/en/latest/plugins/base.html
+Why Oríon?
+==========
+*Effortless to adopt, deeply customizable*
+
+- Adopt it with a single line of code
+- Natively asynchronous. Built for distributed computing
```
Comment (bouthilx) [inline README.rst:60]:
```suggestion
- Natively asynchronous. Easy to distribute and scale.
```
Distributed computing is a specific thing (distributing a simple computation across different process), so this wording is likely to confuse some. Distribute HPO on the other hand should be fine.

File: README.rst [hunk 2]
```diff
@@ -52,31 +52,17 @@ Check out our `pytorch example`_ for a quick demonstration!
...
+- Offers the latest established hyperparameter algorithms
+- Elegant and powerful search-space definitions
```
Comment (bouthilx) [inline README.rst:62]:
```suggestion
- Elegant and rich search-space definitions
```
Maybe? I'm really not sure. Perhaps we should clarify want we want to sell so that we can pick the best adjectives.

**Comment by bouthilx** [comment]
What would you think of forward referencing all the doc in this list? Like

- Adopt it with a single line of code -> point to getting started
- Natively asynchronous. Built for distributed computing -> point to overview
- Offers the latest established hyperparameter algorithms -> point to algo page
- Elegant and powerful search-space definitions -> point to search space page
- Comprehensive configuration system with smart defaults -> point to advanced config page
- Transparent persistence in local or remote database -> point to storage page
- Integrate seamlessly your own hyper-optimization algorithms -> point to plugins page

That may be quite useful.

File: README.rst (hunk 2 above)
Comment (Thomsch) [reply README.rst:62]: I like rich better.

File: README.rst (hunk 1 above)
Comment (Thomsch) [reply README.rst:60]: Thanks for pointing it out. Your suggestion is great. 
I feel it's a tiny bit too wordy compared to the other points. Do you think we could shorten it?

I propose:
> Natively asynchronous and resilient. Distribute and scale unhindered.

**Comment by Thomsch** [comment]
Thanks for the feedback.

This would be a great addition. However, I feel that it would dampen the impact of the sentences if we add a link afterward. Do you have a format in mind to integrate those links?

Alternatively, we could add a link to learn more after the bullet points. The link would redirect them to the getting started where I discuss these points in greater detail and point at the user guide, in the same spirit as scikit's user guide.

What do you think?

File: README.rst (hunk 1 above)
Comment (bouthilx) [reply README.rst:60]: - Natively asynchronous, thus resilient and easy to scale/distribute/paralellize
- Asynchronous, resilient and easy to scale/distribute/paralellize

**Comment by bouthilx** [comment]
I would imagine something like this.
- [Adopt it]() with a single line of code
- [Natively asynchronous](). Built for distributed computing 
- Offers the latest established hyperparameter [algorithms]()
- Elegant and powerful [search-space definitions]()
- Comprehensive [configuration]() system with smart defaults
- Transparent persistence in local or remote [database]()
- [Integrate seamlessly]() your own hyper-optimization algorithms

File: README.rst (hunk 1 above)
Comment (Thomsch) [reply README.rst:60]: I like the first one and agree with parallelize!

**Comment by Thomsch** [comment]
I see. This can work if the pages for each link start with a quick overview/showcase.
